processor/deltatocumulativeprocessor/processor.go

```diff
@@ -29,8 +31,8 @@ type Processor struct {
 	ctx    context.Context
 	cancel context.CancelFunc
 
-	aggr streams.Aggregator[data.Number]
-	exp  *streams.Expiry[data.Number]
+	aggr  streams.Aggregator[data.Number]
+	stale *staleness.Staleness[data.Number]
```

Comment:
minor: You could name this say, `nums` in preparation for having multiple. (IE for histograms, etc)

Reply:
let's do that when its needed. I'd like to experiment with the type system more if we can't just drop the datapoint narrowing on the aggregator altogether